### PR TITLE
Dynamic user permissions

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/7915
Fixes https://github.com/OrchardCMS/OrchardCore/issues/7619

Adds dynamic permissions to managing users, on a per role basis.

Screenshot 2020-12-29 at 11 31 04

### CODE REVIEW

**Comment by deanmarcussen** [comment]
Think I should demo this first.

And if anyone wants to check or question any of the logic (it is complicated), please do ;)

**Comment by scleaver** [comment]
Hey @deanmarcussen - I watched the video on this and had a question. I noticed that if a user had "manage user in roles - editor" it appeared that they couldn't then see any users outside that role. But what if I wanted to add an existing user to the role I manage? I can't see them in the user list as it is not in the role yet? In my systems I have used I a person with such position can still see all users it's just that they can't do anything but add/remove them to a that role.

Typical scenario - I work in corporate public relations and I manage the PR role. While I am away on holidays I want another person in the organisation to be able to edit PR related content.

Maybe I am missing something?

**Comment by scleaver** [comment]
Oh hang on... you are grappling with that at the end of the video... sorry

**Comment by deanmarcussen** [comment]
@scleaver yes, I think changed probably to what you are saying.

- all users in all roles visible
- manage permission gives ability to enable/disable change password etc
- assign roles permission to assign roles

features that can't be done, based on restricted permissions, are greyed out, or not visible. 
e.g. can't assign administrator role, can still assign others, but can't change/remove the administrator role...

bulk actions, and user filters, another pr ;)

File: src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Module/.template.config/template.json [hunk 1]
```diff
@@ -23,8 +23,8 @@
           "description": "Target net5.0"
         }
       ],
-      "replaces": "netcoreapp3.1",
-      "defaultValue": "netcoreapp3.1"
+      "replaces": "$(TemplateTargetPackageFramework)",
```
Comment (deanmarcussen) [inline src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Module/.template.config/template.json:26]: @agriffard something has gone slightly of with the merge here. Can you check that what you expect to be in these template modules is still there ok?

**Pull request merged**